src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -176,36 +160,28 @@ public static string GetFileHash(string path)
         }
 
         /// <summary>
-        /// Update InstallerManifest's Installer nodes based on specified package file paths.
+        /// Update InstallerManifest's Installer nodes using the InstallerUpdateHelpers.
         /// </summary>
+        /// <param name="installerUpdateHelpers">List of <see cref="InstallerUpdateHelper"/>.</param>
         /// <param name="installerManifest"><see cref="InstallerManifest"/> to update.</param>
-        /// <param name="installerUrls">InstallerUrls where installers can be downloaded.</param>
-        /// <param name="paths">Paths to packages to extract metadata from.</param>
-        /// <param name="detectedArchOfInstallers">List of DetectedArch objects that represent each installers detected architectures.</param>
-        /// <param name="newInstallers">List of newly parsed installer nodes.</param>
-        public static void UpdateInstallerNodesAsync(
-            InstallerManifest installerManifest,
-            IEnumerable<string> installerUrls,
-            IEnumerable<string> paths,
-            out List<DetectedArch> detectedArchOfInstallers,
-            out List<Installer> newInstallers)
+        public static void UpdateInstallerNodesAsync(List<InstallerUpdateHelper> installerUpdateHelpers, InstallerManifest installerManifest)
         {
-            var newPackages = paths.Zip(installerUrls, (path, url) => (path, url)).ToList();
-            newInstallers = new List<Installer>();
-            detectedArchOfInstallers = new List<DetectedArch>();
             var existingInstallers = new List<Installer>(installerManifest.Installers);
             List<Installer> unmatchedInstallers = new List<Installer>();
             List<Installer> multipleMatchedInstallers = new List<Installer>();
             List<string> parseFailedInstallerUrls = new List<string>();
 
-            foreach (var (path, url) in newPackages)
+            foreach (var installerUpdateHelper in installerUpdateHelpers)
             {
-                if (!ParsePackageAndGenerateInstallerNodes(path, url, newInstallers, null, ref detectedArchOfInstallers))
+                if (!ParsePackageAndGenerateInstallerNodes(installerUpdateHelper, null))
                 {
-                    parseFailedInstallerUrls.Add(url);
+                    parseFailedInstallerUrls.Add(installerUpdateHelper.InstallerUrl);
                 }
             }
 
+            List<Installer> newInstallers = new List<Installer>();
+            installerUpdateHelpers.ForEach(x => newInstallers.AddRange(x.NewInstallers));
```

Comment:
= installerUpdateHelpers.SelectMany(x => x.NewInstallers).ToList()

Reply:
Actually, if you're just wanting the count, you could do SelectMany(...).Count(), or even installerUpdateHelpers.Sum(x => x.NewInstallers.Count)

Reply:
you are truly a linq magician. Thanks for the suggestion :) simplified to just determining the count of new installers.